### PR TITLE
Final Capstone

### PR DESCRIPTION
**Pull request by Tobinchilongo**
### In this Milestone We:

Completed all the project requirements both in front and on the backend..

**The learning objectives of this project are:**

- [x] Apply technical knowledge and skills gained in previous modules in a complex project.
- [x] Understand pros and cons of different approaches of connecting Ruby on Rails back-end with React front-end.
- [x] Understand principles of Ruby on Rails and React frameworks.
- [x] Apply JavaScript best practices and language style guides in code.
- [x] Apply React best practices and language style guides in code.
- [x] Learn about and practice giving constructive feedback to teammates.
- [x] Perform a code review for a team member.
- [x] Use the "Review change"" feature from GitHub.
- [x] Write clear, professional, and respectful review comments for other team members.
- [x] Explain "why" a change is requested when giving a code review.
- [x] Plan a 2+ week project with no interim Microverse-set milestones and submit it on time.
- [x] Apply knowledge of setting working agreements to set group project teams up for success.
- [x] Independently implement best practices for group agreements to improve teamwork in larger group project.
- [x] Demonstrate ability to apply best practices of communication for resolving teamwork challenges.
- [x] Understand that respect is the foundation of strong relationship-building with teammates.
- [x] Show up throughout group projects as a reliable and committed team member who communicates and manages your time effectively.
- [x] Recognize the importance of investing in good working relationships with teammates.
- [x] Understand principles of strong teamwork (being reliable, committed, and consistent) and how to apply them in group projects.
- [x] Recognize the value of making equal contributions to group projects to produce the best outcome.
- [x] Use empty Kanban board to manage tasks with the team and own time on the project.

Here is the link to our Frontend repo: [Rent-A-Car-Repo](https://github.com/jerryowusu/final-capstone-frontend)

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  it 'Name should not be nil' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'Email should not be nil' do
+    subject.email = nil
+    expect(subject).to_not be_valid
+  end
```
Comment (mcrd25) [inline spec/models/user_spec.rb:17]: [OPTIONAL]
Consider positive and negative scenarios when doing tests. While this is a simple application, doing this in the real world would help catch bugs. What do you expect to happen when negative as well as what do you expect to be output when positive. Consider this for all tests written before merging.

**Review by mcrd25 — APPROVED** [review]
Hi @Tobinchilongo @cynthiainga @jerryowusu , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: spec/models/user_spec.rb (hunk 1 above)
Comment (jerryowusu) [reply spec/models/user_spec.rb:17]: Thanks for drawing our attention to this @mcrd25. We will definitely look into that.

**Comment by jerryowusu** [comment]
Thanks a lot for the review @mcrd25

**Pull request merged**